decred/decred/crypto/secp256k1/curve.py

```diff
@@ -260,8 +270,10 @@ def scalarMult(self, Bx, By, k):
         # elliptic curves states that P(x, y) = -P(x, -y), it's faster and
         # simplifies the code to just make the point negative.
         if k1 < 0:
+            k1 = -k1
```

Comment:
In Go, KoblitzCurve.splitK always returns absolute k1 and k2, and their signs are returned separately. Here, KoblitzCurve.splitK returns signed k1 and k2. When they are negative they cannot be converted to ByteArrays on lines 283-284 below. Their negative sign is handled explicitly here, so they can be turned into absolute values safely.